arangod/Aql/AqlItemBlock.h

```diff
@@ -349,6 +342,35 @@ class AqlItemBlock {
   /// be used to recreate the AqlItemBlock via the Json constructor
   void toVelocyPack(transaction::Methods* trx, arangodb::velocypack::Builder&) const;
 
+  /// @brief test if the given row is a shadow row and conveys subquery
+  /// information only. It should not be handed to any non-subquery executor.
+  bool isShadowRow(size_t row) const {
+    /// This value is only filled for shadowRows.
+    /// And it is guaranteed to be only filled by numbers this way.
+    return _data[getSubqueryDepthAddress(row)].isNumber();
+  }
+
+  AqlValue const& getShadowRowDepth(size_t row) const {
```

Comment:
Yes, but this will be done on the ShadowRow interface.
Here we only work on AQL values on all areas, so it is more convenient to have them unified for serialization / deserialization / copying.

Only the consumer needs to know that it is actually a number type.